### ISSUE TITLE
CA2208 should not report a violation when thrown from a property

### ISSUE THREAD

**Issue by xtqqczze** [dotnet/roslyn-analyzers#5209]
### Analyzer

**Diagnostic ID**: [CA2208](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2208)

### Describe the improvement

Specifying `value` as a `paramName` for `ArgumentException` in a property does not add any value over what the stack trace provides, therefore a violation should not be reported.

#### Existing violation

```csharp
internal string Property
{
    set
    {
        _backing = value ?? throw new ArgumentNullException(); // CA2208 violation
    }
}
```

Creates the message:
```
Value cannot be null.
```

#### Fix of violation

```csharp
internal string Property
{
    set
    {
        _backing = value ?? throw new ArgumentNullException(nameof(value));
    }
}
```

Creates the message:
```
Value cannot be null.
Parameter name: value
```

**Comment by Youssef1313** [comment]
I recall this is something that @bartonjs pointed to in the design discussion of this analyzer. Not sure why the implementation didn't match this decision?

@bartonjs The analyzer currently allows both the implicit `value` parameter and the property name.  Can you please confirm if the analyzer should be completely relaxed in properties (i.e, don't analyze properties completely)?

**Comment by RikkiGibson** [comment]
Perhaps if the code is moved from an accessor to a method then it makes it easier to see that it needs to be changed.

That justification seems a little flimsier, though, assuming that this analyzer is running and will report a warning in that scenario.

**Comment by buyaa-n** [comment]
> I recall this is something that @bartonjs pointed to in the design discussion of this analyzer. Not sure why the implementation didn't match this decision?

I recall that the decision was to use `value`  for properties, not sure what decision are you talking about according to that discussion https://github.com/dotnet/runtime/issues/36517#issuecomment-654411745 this is an expected scenario and it is not a bug

**Comment by Youssef1313** [comment]
@buyaa-n Yeah I see my comment is a bit misleading (specifically the last sentence). However, the analyzer shouldn't allow `nameof(PropertyName)`. So this would be a false negative.

https://github.com/dotnet/runtime/issues/33763#issuecomment-623760581
